sunpy/map/maputils.py

```diff
@@ -1,7 +1,267 @@
 """
-This submodule provides utility functions to act on `sunpy.map.Map` instances.
+This submodule provides utility functions to act on `sunpy.map.GenericMap` instances.
 """
+from itertools import chain, product
 
+import numpy as np
 
-__all__ = []
+import astropy.units as u
+from astropy.coordinates import SkyCoord
 
+from sunpy.coordinates import Helioprojective
+
+__all__ = ['all_pixel_indices_from_map', 'all_coordinates_from_map',
+           'map_edges', 'contains_full_disk',
+           'is_all_off_disk', 'is_all_on_disk', 'contains_limb',
+           'coordinate_is_on_disk', 'on_disk_bounding_coordinates']
+
+
+def all_pixel_indices_from_map(smap):
+    """
+    Returns pixel pair indices of every pixel in a map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    Returns
+    -------
+    `~numpy.array`
+        A `numpy.array` with the all the pixel indices built from the
+        dimensions of the map.
+    """
+    return np.meshgrid(*[np.arange(v.value) for v in smap.dimensions]) * u.pix
+
+
+def all_coordinates_from_map(smap):
+    """
+    Returns the coordinates of every pixel in a map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    Returns
+    -------
+    `~astropy.coordinates.SkyCoord`
+        An array of sky coordinates in the coordinate system "coordinate_system".
+    """
+    x, y = all_pixel_indices_from_map(smap)
+    return smap.pixel_to_world(x, y)
+
+
+def map_edges(smap):
+    """
+    Returns the pixel locations of the edges of an input map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    Returns
+    -------
+    `~dict`
+        Returns the pixels of edge of the map.
+    """
+    # Calculate all the edge pixels
+    nx, ny = smap.dimensions.x.value, smap.dimensions.y.value
+    bottom = list(product([0.0], np.arange(nx))) * u.pix
+    top = list(product([ny - 1], np.arange(nx))) * u.pix
+    lhs = list(product(np.arange(ny), [0])) * u.pix
+    rhs = list(product(np.arange(ny), [nx - 1])) * u.pix
+    return {"top": top, "bottom": bottom, "lhs": lhs, "rhs": rhs}
```

Comment:
I understand what `top` and `bottom` means... well... if we talk about coordinates, if this are pixels I would have my doubts The lower value pixel is shown on the top?. But what does `lhs` or `rhs` mean? `left` and `right`?

Reply:
I have changed the definitions of the coordinates in order to be more consistent with how the user would expect them to be defined.  The function now returns a tuple, though I don't know that this is clearer.  Also, are the meanings of rhs and lhs not understood?